x-pack/test/api_integration/apis/telemetry/telemetry_optin_notice_seen.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { Client, DeleteDocumentParams, GetParams, GetResponse } from 'elasticsearch';
+import { TelemetrySavedObjectAttributes } from '../../../../../src/legacy/core_plugins/telemetry/server/telemetry_repository';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function optInTest({ getService }: FtrProviderContext) {
+  const client: Client = getService('es');
+  const supertest = getService('supertest');
+
+  describe('/api/telemetry/v2/optIn API Telemetry User has seen OptIn Notice', () => {
+    it('should update telemetry setting field via PUT', async () => {
+      await client.delete({
+        index: '.kibana',
+        id: 'telemetry:telemetry',
+      } as DeleteDocumentParams);
+
+      await supertest
+        .put('/api/telemetry/v2/userHasSeenNotice')
+        .set('kbn-xsrf', 'xxx')
+        .expect(200);
+
+      const {
+        _source: { telemetry },
+      }: GetResponse<{
+        telemetry: TelemetrySavedObjectAttributes;
+      }> = await client.get({
+        index: '.kibana',
+        id: 'telemetry:telemetry',
+      } as GetParams);
+
+      expect(telemetry.userHasSeenNotice).to.be(true);
+    });
```

Comment:
The only thing I think the test code is missing is a clean-up step to remove the `telemetry:telemetry` document after the test did its update

Reply:
Also, I didn't see an easy way to test that this API works with a non-privileged user.